### PR TITLE
Post-merge fixes for CORDA-827 and more unit tests

### PR DESCRIPTION
**Pull request by IgorNitto**
* Higher unit test coverage on RPCSecurityManager
* Fixes to corner cases in permission parsing and bogus tryAuthenticate
* Tiny docsite fixes and missing changelog entry

### CODE REVIEW

**Comment by IgorNitto** [comment]
My last planned fix is now on this branch:
 * Revisited docsite page, now part of a new section in 'clientrpc.rst'
 * Renamings of cache config variables: 'capacity' -> 'maxEntries', 'expiryTimeSecs' -> 'expireAfterSecs'

File: docs/source/clientrpc.rst [hunk 1]
```diff
@@ -71,6 +68,136 @@ Fine grained permissions allow a user to invoke a specific RPC operation, or to
 - to invoke a RPC operation: ``InvokeRpc.<rpc method name>`` e.g., ``InvokeRpc.nodeInfo``.
 .. note:: Permission ``InvokeRpc.startFlow`` allows a user to initiate all flows.
 
+RPC security management
+-----------------------
+
+Hard coding user accounts in the `rpcUsers`` field provides a quick way of allowing node's RPC to be accessed by a fixed
```
Comment (sollecitom) [inline docs/source/clientrpc.rst]: Ouch, wrong double ` to close a single one. It messes rendering up.

Comment (IgorNitto) [reply docs/source/clientrpc.rst]: Thanks, fixed in last commit
